make_tag.py

```diff
@@ -34,74 +34,26 @@ def format_version(major, minor, build):
 
 
 def get_next_version(bump_major, bump_minor):
-    # looking to parse tags of form: 'release-0.1.36-cutoff' to get the latest version number and increment build
-    # component.
-    # Because we have some inconsistency in the prefix (sometimes release-, sometimes release-venice-), assume tags are
-    # either of the two:
-    # 1. Full versions (e.g. '0.1.37')
-    # 2. End in '-<version>-cutoff' (e.g. '-0.1.37-cutoff')
-    tags_text = check_output(['git', 'tag'], text=True)  # tag1\ntag2\ntag3...
-    tags_lines = [str(l) for l in tags_text.splitlines()]  # ['tag1', 'tag2', ...]
-
-    full_version_regex = '(^\\d+(?:.\\d+)+$)'
-    legacy_version_regex = '^release-(\\d+(.\\d+)+)-cutoff$'
-    tags_numbers = []
-    for line in tags_lines:
-        full_version_regex_match = re.search(full_version_regex, line)
-        if full_version_regex_match:
-            tags_numbers.append(full_version_regex_match.group(1))
-            continue
-
-        legacy_version_regex_match = re.search(legacy_version_regex, line)
-        if legacy_version_regex_match:
-            tags_numbers.append(legacy_version_regex_match.group(1))
-            continue
-
-    version_numbers = [n.split('.') for n in tags_numbers if len(n.split('.')) == 3]  # [ ['0','1','36'], ...]
-    version_ints = [[int(part) for part in number] for number in version_numbers]  # [[0,1,36], ...]
-
-    max_major = 0
-    for v in version_ints:  # [0, 1, 36]
-        max_major = max(max_major, v[0])
-
-    if bump_major:
-        max_major += 1
-
-    major_filtered_version_numbers = [n for n in version_ints if n[0] == max_major]
+    tag_text = check_output(['git', 'describe', '--tags', '--abbrev=0', 'main'], text=True)
+    tag_text = tag_text.rstrip()
+    commits = check_output(['git', 'log', f'{tag_text}..HEAD', '--oneline'], text=True)
+    if commits == "":
+        print(f'Not pushing new tag as there is no new commit since the last tag {tag_text}')
+        sys.exit()
+    version_numbers = tag_text.split('.')  # ['0','1','36']
+    version_ints = [int(part) for part in version_numbers]  # [0,1,36]
 
+    major = version_ints[0]
     if bump_major:
-        if major_filtered_version_numbers:
-            raise Exception(f'''
-            Something went wrong. Expected version list for major version {max_major}
-            to be empty. Found {major_filtered_version_numbers}. Verify the script.
-            ''')
-
-        return format_version(max_major, 0, 0)
-
-    max_minor = 0
-    for v in major_filtered_version_numbers:  # [0, 1, 36]
-        max_minor = max(max_minor, v[1])
+        major += 1
```

Comment:
We need to `return format_version(major, 0, 0)`. Same for bump minor

Reply:
You can test these locally without the `--no-verify` option. The script will print the tag it is going to register and ask for confirmation